src/main/java/tempest/GUI/DeleteModulePage.java

```diff
@@ -0,0 +1,102 @@
+package tempest.GUI;
+
+import java.awt.event.ActionEvent;
+import java.util.Objects;
+
+import javax.swing.BoxLayout;
+import javax.swing.JButton;
+import javax.swing.JComboBox;
+import javax.swing.JPanel;
+
+import tempest.State;
+import tempest.GUI.components.BackButton;
+import tempest.GUI.components.ModuleDropDown;
+
+public class DeleteModulePage extends Page {
+    private final State state;
+    private final GUIManager manager;
+    private final ModuleDropDown moduleDropDown = new ModuleDropDown();
+
+    private JPanel deleteModulePanel;
+    private JPanel buttonPanel;
+    private JPanel dropDownPanel;
+
+    private BackButton backButton;
+    private JButton deleteButton;
+    private JComboBox<Object> dropDown;
+
+    public DeleteModulePage(State state, GUIManager guiManager) {
+        this.state = state;
+        this.manager = guiManager;
+    }
+
+    @Override
+    public String getName() {
+        return "deleteModulePage";
+    }
+
+    public JPanel getPanel() {
+        deleteModulePanel = new JPanel();
+        buttonPanel = new JPanel();
+        dropDownPanel = new JPanel();
+
+        backButton = new BackButton(manager);
+        deleteButton = new JButton("Delete module");
+
+        deleteButton.addActionListener(this);
+        backButton.addActionListener(this);
+
+        buttonPanel.add(backButton);
+        buttonPanel.add(deleteButton);
+
+        dropDown = moduleDropDown.getModuleDropDown();
+
+        dropDownPanel.add(dropDown);
+
+        deleteModulePanel.add(dropDownPanel);
+        deleteModulePanel.add(buttonPanel);
+
+        deleteModulePanel.setLayout(new BoxLayout(deleteModulePanel, BoxLayout.Y_AXIS));
+
+        return deleteModulePanel;
+    }
+
+    @Override
+    public void actionPerformed(ActionEvent e) {
+        Object source = e.getSource();
+
+        if (source == deleteButton) {
+            handleDeletingSession();
+        }
+    }
+
+    /**
+     * Handles deleting the session and updating the module drop down
+     */
+    private void handleDeletingSession() {
+        // Checks if there are no modules to delete
+        if (dropDown.getItemCount() == 0) {
+            return;
+        }
```

Comment:
I don't think this check is needed with the whole greying out the button thing, 

**it's up to you though** - keeping it means in the future we are able to scrap the greying out thing and not have to rewrite this check.

Reply:
I think this is a good feature - better UX.

Reply:
The link buttons have messed up my tests now.

Reply:
How come? Your getters for the buttons still work right?